File: Dataset/Leetcode/valid/55/110.cpp
class Solution {
public:
    bool XXX(vector<int>& nums) {
        int step = 1;
        for (int i = 0; i < nums.size() - 1; i++) {
            step = max(step-1, nums[i]);
            if (step <= 0) return false;
        }
        return true;
    }
};

File: Dataset/Leetcode/train/7/110.py
class Solution(object):
    def XXX(self, x):
        if x >= 0:
            x = str(x)
            x = x[::-1]
        else:
            x = str(x)
            x = x[1:]
            x = '-' + x[::-1]
        x = float(x)
        if (x < float(-2 << 30)) | (x >= float(2 << 30)):
            return 0
        x = int(x)
        return x

